pkg/server/server.go

```diff
@@ -0,0 +1,691 @@
+	if err != nil {
+		if err == storage.ErrBucketAlreadyExists {
+			s.errorResponse(w, "BucketAlreadyExists", "Bucket already exists", http.StatusConflict)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// handleDeleteBucket handles DeleteBucket operation
+func (s *S3Server) handleDeleteBucket(w http.ResponseWriter, r *http.Request, bucket string) {
+	err := s.storage.DeleteBucket(bucket)
+	if err != nil {
+		if err == storage.ErrBucketNotFound {
+			s.errorResponse(w, "NoSuchBucket", "Bucket does not exist", http.StatusNotFound)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// handleHeadBucket handles HeadBucket operation
+func (s *S3Server) handleHeadBucket(w http.ResponseWriter, r *http.Request, bucket string) {
+	if !s.storage.BucketExists(bucket) {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// handleListObjects handles ListObjects operation
+func (s *S3Server) handleListObjects(w http.ResponseWriter, r *http.Request, bucket string) {
+	query := r.URL.Query()
+	prefix := query.Get("prefix")
+	delimiter := query.Get("delimiter")
+	maxKeys := 1000
+	if mk := query.Get("max-keys"); mk != "" {
+		if parsed, err := strconv.Atoi(mk); err == nil {
+			maxKeys = parsed
+		}
+	}
+	
+	objects, commonPrefixes, err := s.storage.ListObjects(bucket, prefix, delimiter, maxKeys)
+	if err != nil {
+		if err == storage.ErrBucketNotFound {
+			s.errorResponse(w, "NoSuchBucket", "Bucket does not exist", http.StatusNotFound)
+		} else {
+			s.errorResponse(w, "InternalError", err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+	
+	type Contents struct {
+		Key          string    `xml:"Key"`
+		LastModified time.Time `xml:"LastModified"`
+		ETag         string    `xml:"ETag"`
+		Size         int64     `xml:"Size"`
+		StorageClass string    `xml:"StorageClass"`
+	}
+	
+	type CommonPrefix struct {
+		Prefix string `xml:"Prefix"`
+	}
+	
+	type ListBucketResult struct {
+		XMLName        xml.Name       `xml:"ListBucketResult"`
+		Name           string         `xml:"Name"`
+		Prefix         string         `xml:"Prefix"`
+		Delimiter      string         `xml:"Delimiter,omitempty"`
+		MaxKeys        int            `xml:"MaxKeys"`
+		IsTruncated    bool           `xml:"IsTruncated"`
+		Contents       []Contents     `xml:"Contents"`
+		CommonPrefixes []CommonPrefix `xml:"CommonPrefixes,omitempty"`
+	}
+	
+	result := ListBucketResult{
+		Name:        bucket,
+		Prefix:      prefix,
+		Delimiter:   delimiter,
+		MaxKeys:     maxKeys,
+		IsTruncated: false,
+	}
+	
+	for _, obj := range objects {
+		result.Contents = append(result.Contents, Contents{
+			Key:          obj.Key,
+			LastModified: obj.LastModified,
+			ETag:         fmt.Sprintf("\"%s\"", obj.ETag),
```

Comment:
```suggestion
			ETag:         fmt.Sprintf("%q", obj.ETag),
```

This seems to work, and the other modifications as well.